### default slicing for point cloud request
# given relative to robot pelvis
point_cloud:
  min_bounding_box: {x: -3.0, y: -3.0, z: -1.0}
  max_bounding_box: {x:  3.0, y:  3.0, z:  1.0}
  aggregation_size: 2000

### common parameters ##########################################################
# Binary encoded: Possible options: 
# FILTER_NONE                   = 0,
# FILTER_PASS_THROUGH           = 1,
# FILTER_STATISTICAL_OUTLIER    = 2,
# FILTER_VOXEL_GRID             = 4,
# FILTER_MLS_SMOOTH             = 8,
# FILTER_ALL                    = 255
filter_mask: 13

threads: 4

### filter parameters ##########################################################
# params for pass through filter relative to ground level (feet height)
pass_through:
  field_name: z
  min: -1.0
  max:  1.0

# params for voxel grid filter
voxel_grid:
  lx: 0.01
  ly: 0.01
  lz: 2.00

# params for moving least squares smoother
mls_smoother:
  radius: 0.1

# params for statistical outlier filter
statistical_outlier:
  radius: 0.01
  k: 25

# params for normal estimator
normal_estimator:
  radius: 0.05

# params for gradients estimator
gradient_estimator:
  thresh: 0.5

# params for edge detector
edge_detector:
  radius: 0.05
  max_std: 0.022
  non_max_supp_radius: 0.05

# params for grid map generator
grid_map_generator:
  resolution: 0.02
  reconstruct: False

# misc params
misc:
  low_res: 0.05
